File: python/1422. 分割字符串的最大得分.py
class Solution:
    def maxScore(self, s: str) -> int:
        cur = ones = 0
        res = -1
        for i, c in enumerate(s):
            if c == '0':
                cur += 1
            else:
                cur -= 1
                ones += 1
            if i < len(s) - 1:
                res = max(res, cur)
        return res + ones

File: go/general/1422. 分割字符串的最大得分.go
package main

func max(a, b int) int {
	if a > b {
		return a
	} else {
		return b
	}
}

func maxScore(s string) int {
	res := -1
	left := 0
	right := 0
	for i, v := range s {
		if v == '0' {
			left += 1
		} else {
			left -= 1
			right += 1
		}
		if i < len(s)-1 {
			res = max(left, res)
		}
	}
	return res + right
}
